draft-ietf-mls-protocol.md

```diff
@@ -3127,48 +3110,78 @@ struct {
+    MAC confirmation_tag;
+    optional<HPKEPublicKey> external_pub;
     KeyPackageRef signer;
     opaque signature<0..2^16-1>;
-} PublicGroupState;
+} GroupInfo;
 ~~~
 
-Note that the `tree_hash` field is used the same way as in the Welcome message.
-The full tree can be included via the `ratchet_tree` extension
-{{ratchet-tree-extension}}.
+New members MUST verify the `signature` using the public key taken from the
+credential in the leaf node of the member with KeyPackageRef `signer`. The
+signature covers the following structure, comprising all the fields in the
+GroupInfo above `signature`:
 
-The signature MUST verify using the public key taken from the credential in the
-leaf node of the member with KeyPackageRef `signer`. The signature covers the
-following structure, comprising all the fields in the PublicGroupState above
-`signature`:
-
-~~~~~
+~~~
 struct {
-    ProtocolVersion version = mls10;
     CipherSuite cipher_suite;
     opaque group_id<0..255>;
     uint64 epoch;
     opaque tree_hash<0..255>;
-    opaque interim_transcript_hash<0..255>;
+    opaque confirmed_transcript_hash<0..255>;
     Extension group_context_extensions<0..2^32-1>;
     Extension other_extensions<0..2^32-1>;
-    HPKEPublicKey external_pub;
+    MAC confirmation_tag;
+    optional<HPKEPublicKey> external_pub;
     KeyPackageRef signer;
-} PublicGroupStateTBS;
-~~~~~
+} GroupInfoTBS;
+~~~
 
-This signature authenticates the HPKE public key, so that the joiner knows that
-the public key was provided by a member of the group.  The fields that are not
-signed are included in the key schedule via the GroupContext object.  If the
-joiner is provided an inaccurate data for these fields, then its external Commit
-will have an incorrect `confirmation_tag` and thus be rejected.
+The `external_pub` public key is only required for External Commits and can be
+ommitted if External Commits are not allowed due to an application-level policy.
 
-The information in a PublicGroupState is not deemed public in general, but
-applications can choose to make it available to new members in order to allow
-External Commits.
+#### Joining via External Commits
+
+External Commits are a mechanism for new members (external parties that want to
+become members of the group) to add themselves to a group, without requiring
+that an existing member has to come online to issue a Commit that references an
+Add Proposal.
+
+Whether existing members of the group will accept or reject an External Commit
+follows the same rules that are applied to other handshake messages.
+
+New members can create and issue an External Commit if they have access to the
+following information for the group's current epoch:
+
+* group ID
+* epoch ID
+* ciphersuite
+* public tree hash
+* confirmed transcript hash
+* confirmation tag of the most recent Commit
+* group extensions
+* external public key
+
+Thus to join a group via an External Commit, a new member needs a GroupInfo with
+an `external_pub` present.
```

Comment:
Suggest saying a little more about the dynamics here:

```suggestion
In other words, to join a group via an External Commit, a new member needs a GroupInfo with
an `external_pub` present.  Thus, a member of the group can enable new clients to join by
making a GroupInfo object available to them.  Note that because a GroupInfo object is specific 
to an epoch, it will need to be updated as the group advances.  In particular, each GroupInfo
object can be used for one external join, since that external join will cause the epoch to change.
```